### ISSUE TITLE
Glissando only working after inserting or removing blank measure after opening .mscz file.

### ISSUE THREAD

**Issue by MtlMathman** [musescore/MuseScore#23052]
### Issue type

General playback bug

### Bug description

When I open a new score, Electric Bass guitar, Muse Guitars Vol 1 (everything else default).
I entered a glissando between two notes, then deleted the trailing measures. When played, the glissando does not sound. If I then add an empty measure in front, it does. I saved this file. When I reopen, the glissando does not sound. If I delete the leading empty measure, the glissando sounds. It seems that the glissando does not sound when I open the file until I add or delete a leading measure. Also, please note that when I play the one measure score, it plays for three measures.

### Steps to reproduce

Open the .mscz file (in the zip file), play it back. The glissando does not sound.
Delete the first empty measure.
Play it back. The glissando sounds.

Finally, look at the measure counter when playing back. When the score shows 2 measures, it will play for 4. When the score shows 1 measure, it will play for 3.

Please note that I've tried this with LP Electric - Lead as well. Same symptoms.

[Gliss.mscz.zip](https://github.com/user-attachments/files/15522259/Gliss.mscz.zip)

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/170187762/894d10e2-2b15-4281-b1d6-f8355c25a63f

Preview doesn't always seem to work, so here is a zip file with the .mp4 file.

[Glissando screencast capture.zip](https://github.com/user-attachments/files/15522280/Glissando.screencast.capture.zip)

### MuseScore Version

MuseScore Studio 4.3.1.241490902 revision 026c26b

### Regression

I don't know

### Operating system

MacOS Sonoma 14.5 on an M2 2022 MacBook Air

### Additional context

MuseSounds: Muse Guitars Vol 1: Electric Bass

I first encountered this with Musescore Studio version 4.3.0.241231433 Revision 5f36e74. I don't recall glissandos working in earlier versions of MuseScore 3, but I cannot support that statement with evidence.

The included .mp4 screencast capture does have sound, but I cannot hear it on the preview.

**Comment by MtlMathman** [comment]
FYI, I had posted the issue on the community forum earlier, but was not receive any responses that advanced the resolution to the issue.

https://musescore.org/en/node/364419

**Comment by MtlMathman** [comment]
Here is another score. On the same score, a chromatic glissando (measure 2 to 3) does not trigger on playback, yet on the same score, the chromatic glissando (measure 9 to 10) on the same notes do trigger on playback. Both cross a measure boundary. The tempo has been slowed down from 75 bpm to make the glissando more obvious.

https://github.com/musescore/MuseScore/assets/170187762/e6de4d75-bc7b-47d2-8fcb-5df465f1044e

[Europa.mscz.zip](https://github.com/user-attachments/files/15542377/Europa.mscz.zip)

**Comment by MtlMathman** [comment]
Additional testing and information.

I had another user create a blank score (one measure, guitar), and send me the resultant .mscz file. If I build off this file, i.e. adding an electric bass, then deleting the guitar that he added, the glissando works fine, without adding or deleting measures. It appears to be something in my specific installation.

I would like to completely reinstall MuseScore Studio and eliminating all the carried configuration from previous versions, e.g. from MuseScore 3. Is there a way to get a "factory reset" of MuseScore Studio so that no remnants from previous versions are carried over?

**Comment by MarcSabatella** [comment]
MuseScore 4 never carries anything forward from MuseScore 3.  But 4.3 inherits settings etc from 4.2.  “Help / Revert to factory settings” clears that.

**Comment by MtlMathman** [comment]
> MuseScore 4 never carries anything forward from MuseScore 3. But 4.3 inherits settings etc from 4.2. “Help / Revert to factory settings” clears that.

Any way to get rid of the reference to an old VST? Even with a factory reset, it finds it.

**Comment by MarcSabatella** [comment]
What do you mean by “old VST”?  No matter how old it is, if it is installed correctly, it will be found.  If you don’t want to use it, uninstall it.

**Comment by MtlMathman** [comment]
> What do you mean by “old VST”? No matter how old it is, if it is installed correctly, it will be found. RThat’s true regardless of whether you ever had another version of MuseScore install as well. If you don’t want to use the VST at all, uninstall it.

I also have IK Multimedia's Amplitube which I use to record. At one point, I had MuseScore find and incorporate IK Multimedia's SampleTank VSTs because it had a far better guitar rendition. At this point though, I'm trying to get MuseScore to as vanilla a configuration as possible to see if it solves my issue, i.e. without having it load SampleTank, but without wiping out SampleTank for Amplitude. Apparently, there is a reference in the documentation that MuseScore may not integrate newer resources if cache is not cleared out first. I don't see where to clear out the cache, so I'm thinking if I remove the reference to the SampleTank VST, that would eliminate that as a potential conflict.

**Comment by MarcSabatella** [comment]
MuseScore automatically finds correctly-installed VST’s; there shouldn’t have been any special process to “find and incorporate” any given VST.  Unless you are saying it is installed *incorrectly*, and you are employing some sort of special hack to have MuseScore Studio find it anyhow?  Or maybe you mean, in some specific score, you *selected* that VST for use in the playback, using the controls in the Mixer?  That information is saved with the score, just as all other mixer settings like volume fader and pane levels are.

**Comment by MtlMathman** [comment]
It's been a while, but no, I would not have the means nor wherewithal to hack an install.

As for selecting a specific VST for a score: again, no, this would be on a new score where MuseSounds would be selected. In preferences, no path is selected for the VST directory, but an IK Multimedia SampleTank 4 appears and is accessible (though not selected) through the mixer.

It is funny you mention the official Support forum as indeed, I did raise this issue there. The response I received was that their preferred route was to log it here in GitHub so it can be tracked and addressed.

**Comment by MarcSabatella** [comment]
It is correct that you shouldn’t need to specify a non-standard install location for your VST’s.  The spec calls for them to always be installed in a special location that all VST-aware applications already know.  So your VST instruments and effects will always be an available to all applications automatically.  You would only need to specify an alternate location if you are attempting to document the standard and install to a no -standard location.

And yes, again, the forum is the place to ask for help using MuseScore.  Then if others are able to help you refine the steps to reproduce the problem and confirm, at that point it becomes appropriate to open an issue here.

But until it’s clear how to actually reproduce the problem and there are precise steps others can follow to reproduce it themselves, pleas keep the discussion in the forum.  Whoever said asking questions on GitHub was the way to go was mistaken.  GitHub is for bug reports that are confirmed as reproducible given your score and/or precise listed steps.

**Comment by MtlMathman** [comment]
In the initial problem description, I have indeed included the score, outlined the steps to reproduce the problem. can do it every time, and I have included a screencast capture of my doing so.

**Comment by MarcSabatella** [comment]
Indeed; I wasn’t talking about the initial report, which I can confirm is reproducible for me as well.

I was talking about the subsequent discussion about how to reinstall MuseScore, how to reset preferences, how to manage VST’s.  All of that is better suited to the forum, so the issue is just focused on the steps to reproduce the problem, and the developers are not sidetracked by having to wade through a long thread to see if there other relevant info they would need in order to reproduce the problem or understand the cause.

That’s not to say no discussion here is *ever* warranted.  If you later gain some extra insight that you think might be helpful in helping the developers track down the cause of the bug, that’s good info.  So if after further discussion in the forums you learn something that might be relevant by all means come back here to report it.  The idea, again, is to keep this tracker highly focused on information directly relevant to the developers.
